operator/app/src/test/java/org/keycloak/operator/testsuite/integration/KeycloakDeploymentTest.java

```diff
@@ -286,15 +286,36 @@ public void testHostnameStrictDisabled() {
     @Test
     public void testInitialAdminUser() {
         try {
+            var kc = getDefaultKeycloakDeployment();
+            var kcAdminSecret = new KeycloakAdminSecret(k8sclient, kc);
+
+            k8sclient
+                    .resources(Keycloak.class)
+                    .inNamespace(namespace)
+                    .delete();
+            k8sclient
+                    .secrets()
+                    .inNamespace(namespace)
+                    .withName(kcAdminSecret.getName())
+                    .delete();
+
+            // Making sure no other Keycloak pod is still around
```

Comment:
I was thinking about ... the tradeoff here is that by making it generic we risk increasing every test execution time due to the extra checks

Reply:
tentatively keeping this implementation for now, not fussy around changing/improving it, but it can be done in a subsequent iteration. (e.g. I won't have time to test it right now)